### ISSUE TITLE
Transform for grayscale images.

### ISSUE THREAD

**Issue by soumyadeepg** [pytorch/vision#288]
When using RGB images I wrote the transform like

transform_list = [transforms.ToTensor(),
transforms.Normalize((0.5, 0.5, 0.5), (0.5, 0.5, 0.5))]

And it worked perfectly.

But for grayscale images I there is no documentation for transform given on this page http://pytorch.org/docs/master/torchvision/transforms.html. 

Can anyone help me on how to write a transform for grayscale images ??

Thanks in advance.

**Comment by vishwakftw** [comment]
Use this: `transforms.Normalize((0.5, ), (0.5, ))` instead. This is because `grayscale images have only one channel.

**Comment by alykhantejani** [comment]
@vishwakftw is correct. I'll send a PR which adds tests for this and updates the docs to reflect this.

**Comment by samarthsarin** [comment]
> Use this: `transforms.Normalize((0.5, ), (0.5, ))` instead. This is because grayscale images have only one channel.

Why is there a need of , after 0.5 in Normalize?

**Comment by yixuanren** [comment]
> > Use this: `transforms.Normalize((0.5, ), (0.5, ))` instead. This is because grayscale images have only one channel.
> 
> Why is there a need of , after 0.5 in Normalize?

Same question here. And it indeed does not work if lack of comma

**Comment by VinhLoiIT** [comment]
Because Python requires comma to make it a 1-tuple. Without comma, Python will understand it as a singular number (0.5). In the code of Normalization, it requires an N-tuple for N channels, such as a 3-tuple for 3 channels

**Comment by Mayurji** [comment]
But this doesn't solve the issue, I am loading gray scale images with batch size of 16 with dim 150 * 150 and performing the transforms as shown in below images 

![image](https://user-images.githubusercontent.com/10477789/85202559-c1cf3c00-b324-11ea-9a7f-a1a67e58a17f.png)
and i am getting the one instance of dataloader as (16, 3, 100, 100), 100 is because of the resize but channel as 3 ?

![image](https://user-images.githubusercontent.com/10477789/85202584-fd6a0600-b324-11ea-9552-e73c6c0cae2e.png)

**Comment by diechr** [comment]
I think the problem is that ImageFolder loads the images as colour images although they are grayscale.  You can simply add another transformation that makes them grayscale again: `transforms.Grayscale(1),`

**Comment by Mayurji** [comment]
Yes, i have done that and it worked. But i just don't get, why it didn't worked in the first place.

**Comment by fmassa** [comment]
@Mayurji with latest versions of torchvision, if you pass a single value for the `Normalize` function and feed a 3-channel RBG Image, `Normalize` will still work
